### ISSUE TITLE
Output Message - Differenciate the container from its content

### ISSUE THREAD

**Issue by Amachua** [stoplightio/spectral#520]
**User story.**
As a API Designer, When I defined a field that doesn't match a regular expression, Then the value of the property can be prompted in the output.

**Additional information** 

When I use the following OpenAPI specification

```YAML
openapi: 3.0.0

x-format-version: "1.0"

info:
  title: Dummy title
  description: Dummy description
  version: 1.0.0

paths:
  /some/path:
    get:
      description: Dummy description
      parameters:
        - in: header
          name: some_header
          description: Some header description.
          schema:
            type: string
        - in: header
          name: someHeader
          description: Some header description.
          schema:
            type: string
        - in: header
          name: some-header
          description: Some header description.
          schema:
            type: string
      responses:
        "200":
          description: All is good
```
And I use the following rule:
```YAML
  header-parameter-names-kebab-case:
    type: style
    severity: error
    recommended: true
    description: A parameter in the header should be written in kebab-case
    message: "'{{property.value}}' is not kebab-cased: {{error}}"
    given: $..parameters[?(@.in === 'header')]
    then:
      field: name
      function: pattern
      functionOptions:
        match: ^[a-z0-9]+((-[a-z0-9]+)+)?$
```

Then I would like to have as output something like this:
```
   16:17     error  [header-parameter-names-kebab-case] 'some_header' is not kebab-cased: must match the pattern '^[a-z0-9]+((-[a-z0-9]+)+)?$'
      -> paths,/some/path,get,parameters,0,name
   21:17     error  [header-parameter-names-kebab-case] 'someHeader' is not kebab-cased: must match the pattern '^[a-z0-9]+((-[a-z0-9]+)+)?$'
      -> paths,/some/path,get,parameters,1,name
```

At the end I would like to have another property (_property.value_ in the sample) that can be used in the message in addition to the existing ones (property, error and description if I'm not wrong) and that can be use to give back the value of the field (easier for a functional to understand the output message).

If the issue is unclear, please do not hesitate to ask for more information. :)

**Comment by P0lip** [comment]
@Amachua 
> If the issue is unclear, please do not hesitate to ask for more information. :)

Everything is clear, no worries :)
I believe that having an access to value of property would very beneficial for everyone, therefore I'm willing to implement this as a part of the next release. Stay tuned.
